### PR TITLE
[TTS]fastspeech2 conformer

### PR DESCRIPTION
**Pull request by yt605155624**
1. add conformer in fastspeech2
2. rename modules/fastspeech2_transformer -> modules/transformer
3. rename modules/fastspeech2_predictor -> modules/predictor
2. encoder: conformer ,decoder conformer, model is training

### CODE REVIEW

File: paddlespeech/t2s/modules/conformer/convolution.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        Returns
+        ----------
+        paddle.Tensor
+            Output tensor (#batch, time, channels).
+        """
+        # exchange the temporal dimension and the feature dimension
+        x = x.transpose([0, 2, 1])
+
+        # GLU mechanism
+        x = self.pointwise_conv1(x)  # (batch, 2*channel, dim)
```
Comment (zh794390558) [inline paddlespeech/t2s/modules/conformer/convolution.py]: shape信息错了。

File: paddlespeech/t2s/modules/conformer/encoder_layer.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            self_attn,
+            feed_forward,
+            feed_forward_macaron,
+            conv_module,
+            dropout_rate,
+            normalize_before=True,
+            concat_after=False,
+            stochastic_depth_rate=0.0, ):
+        """Construct an EncoderLayer object."""
+        super(EncoderLayer, self).__init__()
```
Comment (zh794390558) [inline paddlespeech/t2s/modules/conformer/encoder_layer.py]: super().__init__()

File: paddlespeech/t2s/modules/nets_utils.py [hunk 3]
```diff
@@ -150,3 +150,17 @@ def initialize(model: nn.Layer, init: str):
                                               nn.initializer.Constant())
     else:
         raise ValueError("Unknown initialization: " + init)
+
+
+def get_activation(act):
```
Comment (zh794390558) [inline paddlespeech/t2s/modules/nets_utils.py]: 可以统一到activition里。

File: paddlespeech/t2s/modules/transformer/embedding.py [hunk 4]
```diff
@@ -128,14 +128,87 @@ def forward(self, x):
...
+        Embedding dimension.
+    dropout_rate : float
+        Dropout rate.
+    max_len : int
+        Maximum input length.
+    """
+
+    def __init__(self, d_model, dropout_rate, max_len=5000, dtype="float32"):
+        """Construct an PositionalEncoding object."""
+        super(RelPositionalEncoding, self).__init__()
```
Comment (zh794390558) [inline paddlespeech/t2s/modules/transformer/embedding.py]: init

File: paddlespeech/t2s/modules/transformer/embedding.py [hunk 5]
```diff
@@ -128,14 +128,87 @@ def forward(self, x):
...
+        paddle.Tensor
+            Encoded tensor (batch, time, `*`).
         """
         self.extend_pe(x)
         T = paddle.shape(x)[1]
         x = x + self.alpha * self.pe[:, :T]
         return self.dropout(x)
+
+
+class RelPositionalEncoding(paddle.nn.Layer):
```
Comment (zh794390558) [inline paddlespeech/t2s/modules/transformer/embedding.py]: nn.Layer

File: paddlespeech/t2s/modules/transformer/subsampling.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        Output dimension.
+    dropout_rate : float
+        Dropout rate.
+    pos_enc : paddle.nn.Layer
+        Custom position encoding layer.
+    """
+
+    def __init__(self, idim, odim, dropout_rate, pos_enc=None):
+        """Construct an Conv2dSubsampling object."""
+        super(Conv2dSubsampling, self).__init__()
```
Comment (zh794390558) [inline paddlespeech/t2s/modules/transformer/subsampling.py]: init

File: paddlespeech/t2s/modules/conformer/convolution.py (hunk 1 above)
Comment (yt605155624) [reply paddlespeech/t2s/modules/conformer/convolution.py]: doen

File: paddlespeech/t2s/modules/nets_utils.py (hunk 3 above)
Comment (yt605155624) [reply paddlespeech/t2s/modules/nets_utils.py]: done

File: paddlespeech/t2s/modules/transformer/embedding.py (hunk 5 above)
Comment (yt605155624) [reply paddlespeech/t2s/modules/transformer/embedding.py]: done

File: paddlespeech/t2s/modules/transformer/subsampling.py (hunk 6 above)
Comment (yt605155624) [reply paddlespeech/t2s/modules/transformer/subsampling.py]: done

File: paddlespeech/t2s/modules/conformer/encoder_layer.py (hunk 2 above)
Comment (yt605155624) [reply paddlespeech/t2s/modules/conformer/encoder_layer.py]: done

File: paddlespeech/t2s/modules/transformer/encoder.py [hunk 7]
```diff
@@ -120,7 +120,7 @@ def __init__(self,
                  stochastic_depth_rate: float=0.0,
                  intermediate_layers: Union[List[int], None]=None,
                  encoder_type: str="transformer"):
-        """Construct an Encoder object."""
+        """Construct an Bae Encoder object."""
```
Comment (zh794390558) [inline paddlespeech/t2s/modules/transformer/encoder.py]: base

**Review by zh794390558 — APPROVED** [review]
LGTM

**Pull request merged**